Structure_Engine/Create/Loads/AreaDifferentialTemperatureLoad.cs

```diff
@@ -0,0 +1,101 @@
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using BH.oM.Geometry;
+using BH.oM.Structure.Loads;
+using BH.oM.Base;
+using BH.oM.Structure.Elements;
+using System.Collections.Generic;
+using System.Linq;
+using System;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Creates a differential temperature load to be applied to area elements such as Panels and FEMeshes.")]
+        [InputFromProperty("loadcase")]
+        [Input("positions", "The parametric distance distance from the top (local z) or left (local y) of the property.")]
+        [Input("temperatures", "The temperature at the corresponding position on the surface property.")]
+        [Input("objects", "The collection of elements the load should be applied to.")]
+        [Input("name", "The name of the created load.")]
+        [Output("areaDiffTempLoad", "The created AreaDifferentialTempratureLoad.")]
+        public static AreaDifferentialTemperatureLoad AreaDifferentialTemperatureLoad(Loadcase loadcase, List<double> positions, List<double> temperatures, IEnumerable<IAreaElement> objects, string name = "")
+        {
+            //Checks for positions and profiles
+            if (positions.Count != temperatures.Count)
+            {
+                Reflection.Compute.RecordError("Number of positions and temperatures provided are not equal");
+                return null;
+            }
+            else if (positions.Exists((double d) => { return d > 1; }) || positions.Exists((double d) => { return d < 0; }))
+            {
+                Reflection.Compute.RecordError("Positions must exist between 0 and 1 (inclusive)");
+                return null;
+            }
+
+            List<double> sortedPositions = positions;
+            sortedPositions.Sort();
+
+            if (!positions.SequenceEqual(sortedPositions))
+            {
+                Reflection.Compute.RecordError("Positions must be sorted in ascending order.");
+                return null;
+            }
```

Comment:
This will sort both of the lists, positions as well as sorted positions (line 62 will not copy the list, but point to the same list), which means the check on line 65 will always be true.

Could maybe change to somthing like:

```suggestion
            if (positions.Zip(positions.Skip(1), (a, b) => new { a, b }).Any(p => p.a > p.b))
            {
                Reflection.Compute.RecordError("Positions must be sorted in ascending order.");
                return null;
            }
```

work instead?